src/prices/updater.ts

```diff
@@ -146,26 +156,42 @@ export function createPriceUpdater(
 
           const newPrices: Price[] = [];
 
-          for (const [timestamp, price] of prices) {
+          for (const [timestampMs, price] of prices) {
             try {
-              const blockNumber = await getBlockFromTimestamp(
-                timestamp,
-                0,
-                lastBlockNumber,
-                getBlockTimestamp,
-                logger
-              );
+              const timestampInSeconds = Math.floor(timestampMs / 1000);
+
+              let blockNumber =
+                timestampToBlockMap.get(timestampInSeconds) ?? null;
+
+              if (!blockNumber) {
```

Comment:
This will evaluate to `true` both when `blockNumber`is missing from the map, and in case `blockNumber` is 0 (assuming that can ever happen).

Reply:
good catch